lib/rspec/core/hooks.rb

```diff
@@ -503,6 +503,8 @@ def all_hooks_for(position, scope)
         end
 
         def run_owned_hooks_for(position, scope, example_or_group)
+          return if example_or_group.class.respond_to?(:metadata) &&
+                      example_or_group.class.metadata.keys.include?(:skip)
```

Comment:
Thank you for such kind, detailed feedback! It's always so helpful!!

I'll revise accordingly and push something back up in a second.